modules/back-end/src/Domain/SemanticPatch/FlagComparer.cs

```diff
@@ -0,0 +1,440 @@
+using System.Text.Json;
+using Domain.FeatureFlags;
+using Domain.Targeting;
+
+namespace Domain.SemanticPatch;
+
+public class FlagComparer
+{
+    private readonly FeatureFlag _original;
+    private readonly FeatureFlag _current;
+
+    public FlagComparer(FeatureFlag original, FeatureFlag current)
+    {
+        _original = original;
+        _current = current;
+    }
+
+    public IEnumerable<FlagInstruction> Compare()
+    {
+        var instructions = new List<FlagInstruction>();
+
+        instructions.Add(CompareStatus());
+        instructions.Add(CompareArchived());
+
+        instructions.Add(CompareName());
+        instructions.Add(CompareDescription());
+        instructions.AddRange(CompareTags());
+
+        instructions.Add(CompareVariationType());
+        instructions.AddRange(CompareVariations());
+
+        instructions.Add(CompareDisabledVariation());
+
+        instructions.AddRange(CompareDefaultVariation());
+        instructions.AddRange(CompareTargetUsers());
+        instructions.AddRange(CompareRules());
+
+        return instructions;
```

Comment:
I think you should filter out the noop instruction here

Reply:
resolved in [a9ffb7b](https://github.com/featbit/featbit/pull/467/commits/a9ffb7b10daf7c859671510edaa5d809250ccc7b)